ql/src/java/org/apache/hadoop/hive/ql/processors/CommandProcessorFactory.java

```diff
@@ -173,7 +175,38 @@ private static void setUserSpecificProperties(HiveConf conf) {
     }
   }
 
-  private static void setPropertiesHelper(FDPGatewayBoxConfiguration fdpGatewayBoxConfiguration, HiveConf conf) throws BillingOrgNotFoundException{
+  private static void setPropertiesHelper(FDPGatewayBoxConfiguration fdpGatewayBoxConfiguration, HiveConf conf) throws BillingOrgNotFoundException, IOException, InterruptedException {
+    setQueue(fdpGatewayBoxConfiguration, conf);
+    setJobName(conf);
+  }
+
+  private static void setJobName(HiveConf conf) throws IOException, InterruptedException {
+    String queryId = conf.getVar(HiveConf.ConfVars.HIVEQUERYID);
+    String loggedInUser = QueueFetcher.getLoggedInUser();
+    String mapredJobName = queryId + "-" + loggedInUser;
+    LOG.info("Setting property {} as {}", MAPRED_JOB_NAME, mapredJobName);
+    conf.set(MAPRED_JOB_NAME, mapredJobName);
+    LOG.info("Setting property {} as {}", HIVE_QUERY_NAME, mapredJobName);
+    conf.set(HIVE_QUERY_NAME, mapredJobName);
+  }
+
+  /**
+   * Since multiple commands can be used in session we need to check if we have already set the logged in user name
+   * @param existingPropertyVal
+   * @param loggedInuser
+   * @return
+     */
+  private static String checkExistingNameAndAppendUserIfNotAppended(String existingPropertyVal, String loggedInuser) {
+    String[] splitOfJobName = existingPropertyVal.split("-");
+    if(splitOfJobName[splitOfJobName.length -1].equals(loggedInuser)){
+      return existingPropertyVal;
+    }
+    else {
+      return existingPropertyVal + "-" + loggedInuser;
```

Comment:
In case of ad-hoc queries the existing property value might be huge. Make in logged_inuser + "_<somebetterseperatorstring>-" + existingPropertyval